### PR TITLE
Allow geometry_type to be None at the type level to match documentation

### PR DESCRIPTION
**Pull request by logankaser**
### Description
Allow geometry_type to be None at the type level to match documentation.
Fixes a common type checking error when using geoalchemy

### Checklist

This pull request is:

- [ ] A documentation / typographical error fix
	- Good to go, no issue or tests are needed
- [x] A short code fix
	- [ ] Please include: `Fixes: #<issue number>` in the description if it solves an existing issue
	  (which must include a complete example of the issue).
	- [ ] Please include tests that fail with the `main` branch and pass with the provided fix.
- [ ] A new feature implementation or update an existing feature
	- [ ] Please include: `Fixes: #<issue number>` in the description if it solves an existing issue
	  (which must include a complete example of the feature).
	- [ ] Please include tests that cover every lines of the new/updated feature.
	- [ ] Please update the documentation to describe the new/updated feature.

### CODE REVIEW

**Comment by adrien-berchet** [comment]
Hi @logankaser 
Thank you very much for this work!
The other parameters of this function don't cause any issue? Because there are other parameters that are optional so I'm wondering.

**Comment by logankaser** [comment]
I think we just were not using the other arguments on my team, hopefully these types are accurate as
I've not used those arguments.

File: geoalchemy2/types/__init__.py [hunk 1]
```diff
@@ -114,14 +115,14 @@ class _GISType(UserDefinedType):
...
-        geometry_type="GEOMETRY",
+        geometry_type: Optional[str] = "GEOMETRY",
         srid=-1,
         dimension=2,
         spatial_index=True,
         use_N_D_index=False,
-        use_typmod=None,
-        from_text=None,
-        name=None,
+        use_typmod: Optional[Any] | Literal[False] = None,
```
Comment (adrien-berchet) [inline geoalchemy2/types/__init__.py:123]: Why not just `Optional[bool]` here?

**Comment by adrien-berchet** [comment]
> I think we just were not using the other arguments on my team, hopefully these types are accurate as I've not used those arguments.

Ok, I see :)
Thanks for the new change!

File: geoalchemy2/types/__init__.py (hunk 1 above)
Comment (logankaser) [reply geoalchemy2/types/__init__.py:123]: Technically, it's more accurate, but I did not realize it is not available in `3.7`, bool it is!

**Comment by adrien-berchet** [comment]
Thank you very much!
I'm going to push a new release to make these change available.

**Comment by adrien-berchet** [comment]
The new release is available: https://pypi.org/project/GeoAlchemy2/0.14.7

**Pull request merged**